src/executor_v2/mod.rs

```diff
@@ -244,16 +265,19 @@ impl<S: Storage> Builder<S> {
                 right_types: self.plan_types(right).to_vec(),
             }
             .execute(self.build_id(left), self.build_id(right)),
-
-            HashJoin([op, lkeys, rkeys, left, right]) => HashJoinExecutor {
-                op: self.node(op).clone(),
-                left_keys: self.resolve_column_index(lkeys, left),
-                right_keys: self.resolve_column_index(rkeys, right),
-                left_types: self.plan_types(left).to_vec(),
-                right_types: self.plan_types(right).to_vec(),
+            HashJoin([op, lkeys, rkeys, left, right]) => {
+                if matches!(self.node(op), Expr::Inner) {
+                    hash_join_operator!(self, JoinType::Inner, op, lkeys, rkeys, left, right)
+                } else if matches!(self.node(op), Expr::LeftOuter) {
```

Comment:
Use `match` instead of `if matches!`? (It's surprising that there's no clippy warning? 🤔)

BTW just a personal preference not a requirement: I would define the macro just here, and catch local variables directly without passing arguments. 🤪